### DISCUSSION TITLE
Minta saran dari mentor

### DISCUSSION THREAD

**Discussion by naturallyfunny** [teamalgoritma/community#927] [Q&A] [answered]
Saya lagi ikut lomba AI yang datanya data sentiment qatar airlines. Output lombanya adalah infografis insight dari hasil analisa data yang ada. Saya ingin meminta saran analisa insight apa yang bagusnya saya berikan, informasi apa yang bagusnya saya hasilkan dari data ini agar infografis saya menarik perhatian juri, baik dari analisa, visualisasi, hingga modeling. Mentor-mentor yang sudah berpengelaman sebagai data scientist pastinya jauh lebih hebat dari saya dalam menarik insight. Datasetnya saya lampirkan, belum sepenuhnya dibersihkan.

[cleaned_review_body_title_rating.csv](https://github.com/user-attachments/files/17648380/cleaned_review_body_title_rating.csv)

**Comment by adizaen** [comment, marked as answer]
Dear Pak @ardianthescientist 

Terima kasih banyak Pak atas pertanyaannya. Tetapi kami mohon maaf, kami tidak bisa melakukan pengolahan data yang Bapak berikan. Tetapi kami dapat memberikan beberapa insight yang bisa Bapak pertimbangkan agar dapat membuat infografis yang menarik dan informatif.  Berikut beberapa saran yang dapat kami sampaikan:

**1. Analisis Sentimen (Klasifikasi)**

Bapak bisa lakukan analisis sentimen berdasarkan review dari penupang. Buat model machine learning yang dapat memprediksi apakah review tersebut termasuk ke dalam kategori positif/negatif/netral. 

**2. Analisis Sentimen Berdasarkan Waktu**

Tampilkan bagaimana sentimen terhadap Qatar Airways berubah dari waktu ke waktu. Apakah ada periode tertentu di mana banyak keluhan atau pujian? Misalnya, apakah ada perubahan sentimen yang signifikan terkait dengan perubahan kebijakan perusahaan, pandemi, atau musim liburan? Bisa juga dibuat heatmap yang menunjukkan pola sentimen pada hari tertentu atau jam tertentu dalam seminggu.

**3. Analisis Sentimen Berdasarkan Kategori Layanan**
Misalnya, analisis perbandingan sentimen terhadap aspek-aspek layanan seperti penerbangan, kursi, makanan, pelayanan staf, atau harga tiket. Ini akan memberikan wawasan tentang area mana yang harus ditingkatkan atau dipromosikan. Gunakan teknik topic modeling seperti LDA (_Latent Dirichlet Allocation_) untuk menemukan topik-topik yang sering muncul dalam ulasan terkait Qatar Airways, dan bagaimana sentimen terkait topik tersebut. Misalnya, "makanan di pesawat" atau "kebersihan kabin".

Terkait pengolahan data sentimen, berikut beberapa langkah umum yang bisa dilakukan:
1.	Case-folding: menyeragamkan teks review ke lowercase atau uppercase
2.	Remove numbers: menghapus angka-angka
3.	Remove stopwords: menghapus kata - kata yang tidak bermakna
4.	Remove punctuation: menghapus tanda baca
5.	Stemming: mengubah ke bentuk kata dasar
6.	Remove white space: menghapus spasi
7.	Document-Term Matrix (DTM): mengubah ke bentuk numerik agar bisa dilakukan pemodelan
8.	Remove Infrequent Words: mengurangi komputasi komputer dengan menghapus kata-kata yan jarang muncul.
9.	Modeling: membuat model machine learning untuk melakukan sentimen

Semoga membantu Pak Ardian.


Best regards,
Team Algoritma
